sys/net/gnrc/application_layer/uhcpc/gnrc_uhcpc.c

```diff
@@ -0,0 +1,125 @@
+#include "net/fib.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/ipv6/nc.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netif.h"
+#include "net/ipv6/addr.h"
+#include "net/netdev2.h"
+#include "net/netopt.h"
+
+#include "net/uhcp.h"
+#include "log.h"
+#include "fmt.h"
+
+static kernel_pid_t gnrc_border_interface;
+static kernel_pid_t gnrc_wireless_interface;
+
+static void set_interface_roles(void)
+{
+    kernel_pid_t ifs[GNRC_NETIF_NUMOF];
+    size_t numof = gnrc_netif_get(ifs);
+    int res;
+    uint16_t dev_type;
+
+    for (size_t i = 0; i < numof && i < GNRC_NETIF_NUMOF; i++) {
+        kernel_pid_t dev = ifs[i];
+        res = gnrc_netapi_get(dev, NETOPT_DEVICE_TYPE, 0, &dev_type, sizeof(dev_type));
+        if (res <= 0) {
+            dev_type = NETDEV2_TYPE_UNKNOWN;
+        }
+        if ((!gnrc_border_interface) && (dev_type == NETDEV2_TYPE_ETHERNET)) {
+            ipv6_addr_t addr, defroute;
+            gnrc_border_interface = dev;
+
+            ipv6_addr_from_str(&addr, "fe80::2");
```

Comment:
maybe it's more efficient to use the static initializer `IPV6_ADDR_LINK_LOCAL_PREFIX` and then just modify the last byte for `fe80::2` and `fe80::1` below?